src/main/scala/rise/core/DSL/infer.scala

```diff
@@ -136,9 +139,8 @@ object infer {
       val env1 : Map[String, Type] = env + (tx.name -> tx.t)
       val (te, cs, ftvE) = constrainTypes(env1)(e)
       val ft = FunType(tx.t, te.t)
-      val exprT = genType(expr)
-      val c = TypeConstraint(exprT, ft)
-      (Lambda(tx, te)(ft), cs :+ c, ftvE)
+      val cs1 = constrIfTyped(expr.t)(TypeConstraint(expr.t, ft))
+      (Lambda(tx, te)(ft), cs :++ cs1, ftvE)
```

Comment:
minor style: we usually use `++` instead of `:++`, I think that they do the same.

Reply:
Fixed